dashboard/combined_data.py

```diff
@@ -26,53 +29,62 @@ def download_csv_from_s3_to_dataframe(bucket_name, folder_name, file_name):
 
     try:
         s3_client.download_file(bucket_name, object_key, temp_file_path)
-        print(f"File downloaded successfully: {temp_file_path}")
-        
+        logging.info("File downloaded successfully: %s", temp_file_path)
+
         df = pd.read_csv(temp_file_path)
-        print("File loaded into Pandas DataFrame successfully.")
-        
+        logging.info("File loaded into Pandas DataFrame successfully.")
+
         os.remove(temp_file_path)
-        print("Temporary file removed.")
-        
+        logging.info("Temporary file removed.")
+        df['date_and_hour'] = pd.to_datetime(df['date_and_hour'])
         return df
     except Exception as e:
-        print(f"An error occurred: {e}")
+        logging.error("An error occurred: %s", e)
         return None
 
+
 def fetch_keyword_recordings_as_dataframe():
-    
+
     try:
         connection = get_connection()
-        
+
         query = f"""
         SET search_path TO {ENV["SCHEMA_NAME"]};
         SELECT keyword_recordings_id, keywords_id, total_mentions, avg_sentiment, date_and_hour
         FROM keyword_recordings;
         """
-        
+
         dataframe = pd.read_sql_query(query, connection)
         return dataframe
 
     except psycopg2.Error as e:
-        print(f"Database error: {e}")
+        logging.error("Database error: %s", e)
         return None
     finally:
         if connection:
             connection.close()
 
-if __name__ == "__main__":
+
+def main_combine() -> pd.DataFrame:
+    """Main function to produce dataframe from S3 bucket."""
     load_dotenv()
-    
+
     BUCKET_NAME = ENV["S3_BUCKET_NAME"]
     FOLDER_NAME = ENV["S3_FOLDER_NAME"]
     FILE_NAME = ENV["S3_FILE_NAME"]
 
-    s3_df = download_csv_from_s3_to_dataframe(BUCKET_NAME, FOLDER_NAME, FILE_NAME)
+    s3_df = download_csv_from_s3_to_dataframe(
+        BUCKET_NAME, FOLDER_NAME, FILE_NAME)
     db_df = fetch_keyword_recordings_as_dataframe()
-    
+
     if s3_df is not None and db_df is not None:
         combined_df = pd.concat([s3_df, db_df], ignore_index=True)
-        print("Combined DataFrame:")
-        print(combined_df)
+        logging.info("Combined DataFrame from S3 created successfully.")
+        return combined_df
     else:
-        print("Could not combine DataFrames due to missing data.")
+        logging.ERROR("Could not combine DataFrames due to missing data.")
+        return None
+
+
+if __name__ == "__main__":
+    print(main_combine())
```

Comment:
Should we be logging here or is there no reason as it sits within if __name__ == "__main__"?

Reply:
it is only there to check the result! will remove